packages/dev/core/src/Shaders/ShadersInclude/pbrBlockReflectivity.fx

```diff
@@ -123,17 +121,40 @@ reflectivityOutParams reflectivityBlock(
             // Compute the converted reflectivity.
             surfaceReflectivityColor = mix(0.16 * reflectance * reflectance, baseColor, metallicRoughness.r);
         #else
-            vec3 metallicF0 = metallicReflectanceFactors.rgb;
+            float specularWeight = metallicReflectanceFactors.a;
+            float dielectricF0 = reflectivityColor.a * specularWeight;
+            surfaceReflectivityColor = metallicReflectanceFactors.rgb;
 
             #if DEBUGMODE > 0
-                outParams.metallicF0 = metallicF0;
+                outParams.metallicF0 = vec3(dielectricF0) * surfaceReflectivityColor;
             #endif
 
-            // Compute the converted diffuse.
-            outParams.surfaceAlbedo = mix(baseColor.rgb * (1.0 - metallicF0), vec3(0., 0., 0.), metallicRoughness.r);
-
-            // Compute the converted reflectivity.
-            surfaceReflectivityColor = mix(metallicF0, baseColor, metallicRoughness.r);
+            #ifdef LEGACY_SPECULAR_ENERGY_CONSERVATION
+                outParams.surfaceAlbedo = baseColor.rgb * (1.0 - dielectricF0) * (1.0 - outParams.metallic);

```

Comment:
```suggestion
                outParams.surfaceAlbedo = baseColor.rgb * (vec3(1.0) - vec3(dielectricF0) * surfaceReflectivityColor) * (1.0 - outParams.metallic);
```
because `metallicF0 = vec3(dielectricF0) * surfaceReflectivityColor`